library/model/book.py

```diff
@@ -127,7 +95,7 @@ def from_book(cls, book: Book) -> "BorrowedBook":
 
     def renew_rental(self) -> "BorrowedBook":
         self.due_date += timedelta(days=7)
-        self.current_fee += self.get_weekly_fee()
+        self.current_fee += self.book.get_weekly_fee()
```

Comment:
BorrowedBook seems to have no attribute self.book - did you want to implement that?